src/random-seed/random-seed.c

```diff
@@ -131,8 +133,8 @@ int main(int argc, char *argv[]) {
         if ((uint64_t) st.st_size > buf_size)
                 buf_size = MIN(st.st_size, POOL_SIZE_MAX);
 
-        buf = malloc(buf_size);
-        if (!buf) {
+        info = malloc(sizeof(*info) + buf_size);
```

Comment:
should be: 

```c
info = malloc(offsetof(struct rand_pool_info, buf) + buf_size);
```

Reply:
So, in this case offsetof() and sizeof() are the same, so I won't bikeshed too much over this. But FTR, the reason I prefer the sizeof() idiom is that it ensures one allocates a full struct, even if buf_size is small. Consider something like `struct { long a; char b; char c[]; }`. Allocating using offsetof is error-prone if the allocation is hit with a memset(..., sizeof()), or one uses struct assignment. The fully correct thing is to do the offsetof thing, then max() with sizeof(*info), but that's so much of a mouthful that it requires a helper macro.

Reply:
Well, the way I understood C sizeof() and offsetof(…) will actually result in the very same results in cases like this. however, one makes very clear what is going on here I think for the reader. hence, this is really about readability, nothing else...